packages/core/src/Util/Reachability.ts

```diff
@@ -15,12 +18,28 @@ export default class ReachabilityNavigator implements Reachability {
 			window.addEventListener('online', notifyOnline);
 			window.addEventListener('offline', notifyOffline);
 
+			this._observer = observer;
+			ReachabilityNavigator._observers.push(observer);
+
 			return () => {
 				window.removeEventListener('online', notifyOnline);
 				window.removeEventListener('offline', notifyOffline);
+				ReachabilityNavigator._observers = ReachabilityNavigator._observers.filter(
+					_observer => _observer !== this._observer
+				);
 			};
 		});
 	}
+
+	// expose observers to simulate offline mode for integration testing
+	private static _observerOverride(status: NetworkStatus): void {
+		for (const _observer of ReachabilityNavigator._observers) {
+			if (!(_observer.next && typeof _observer.next === 'function')) {
+				continue;
+			}
```

Comment:
Since we re the only ones adding observers to `ReachabilityNavigator._observers`, I think the only check we need to do is that the observable is still open:

```suggestion
			if (!_observer.closed)) {
				// This would be a good place and time to remove it from ReachabilityNavigator._observers
				continue;
			}
```

Reply:
BTW, my `if` check is wrong, we should continue if it is closed